### DISCUSSION TITLE
is the pypi project official?

### DISCUSSION THREAD

**Discussion by braindevices** [newren/git-filter-repo#596] [Q&A]
https://pypi.org/project/git-filter-repo/

is the pip install officially supported or recommended? if yes why it is not in the Install.md?

**Comment by newren** [comment]
pip used to be mentioned in the INSTALL.md file.  I ripped it out because:

- `pip[3] install git-filter-repo` would not mark git-filter-repo as executable
- pip would not put git-filter-repo in $PATH
- pip, on windows, would sometimes put a broken shebang for the first line of git-filter-repo
- (I also wonder if pip correctly handled symlinks on Windows; I can't remember and can't find comments either way right now.)

Given that installing git-filter-repo (for most uses) consists of downloading a _single_ (executable) file and sticking it in your $PATH, I could realistically say that nothing was better than pip.  By which I mean that providing no special installation method whatsoever was literally an improvement upon what pip was doing -- users could install faster and with less manual work from scratch than by trying to pick up where pip left off.

pipx (which I was unaware of at the time) solves at least the first issue, and might solve the second.  No idea if it's affected by the third or fourth issues, but I have no Windows test environment, and I'm rather unmotivated to open potential floodgates of bugreports when some alternative installation instructions don't measure up.  My experience is that well-meaning installation instruction tweaks work for some people, but surprisingly don't work for many others and those others then complain about how hard it is to install a project whose installation instructions are to download a single executable file and stick it in your path.  Explaining how to download a single executable file and stick it in your PATH seems to be far superior at deflecting the floodgates of those bug reports.
